### PR TITLE
setting read/write http timeouts

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: this pr sets read and write http timeouts. As by default these timeouts are off, it means that clients can hold up connections indefinitely this in turn can potentially use up all available connections in a pool.

### CODE REVIEW

File: cmd/controller/main.go [hunk 1]
```diff
@@ -362,8 +362,10 @@ func createUtilHttpServer(kubeClient *kubernetes.Clientset, kubeconfigProvider c
 	m.Handle("/ready", http.HandlerFunc(health.ReadyEndpoint))
 
 	return &http.Server{
-		Addr:    listenAddress,
-		Handler: m,
+		Addr:         listenAddress,
+		Handler:      m,
+		ReadTimeout:  5 * time.Second,
+		WriteTimeout: 10 * time.Second,
```
Comment (p0lyn0mial) [inline cmd/controller/main.go]: set arbitrary.

**Comment by p0lyn0mial** [comment]
I'll follow up that change to `kubermatic` as well.

**Comment by mrIncompetent** [comment]
@p0lyn0mial can we merge this?

**Comment by p0lyn0mial** [comment]
yes, but it lacks a proper tag from a code owner. please tag the pull.

**Comment by mrIncompetent** [comment]
@p0lyn0mial sorry, missed that

**Pull request merged**